x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/CrossClusterSubqueryIT.java

```diff
@@ -0,0 +1,696 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.action;
+
+import org.elasticsearch.action.admin.indices.alias.IndicesAliasesRequest;
+import org.elasticsearch.action.admin.indices.alias.IndicesAliasesRequestBuilder;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.index.query.RangeQueryBuilder;
+import org.elasticsearch.xpack.esql.VerificationException;
+import org.junit.Before;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import java.util.Locale;
+
+import static org.elasticsearch.core.TimeValue.timeValueSeconds;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.elasticsearch.xpack.esql.EsqlTestUtils.getValuesList;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.greaterThanOrEqualTo;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.hasSize;
+
+// @TestLogging(value = "org.elasticsearch.xpack.esql.session:DEBUG", reason = "to better understand planning")
+public class CrossClusterSubqueryIT extends AbstractCrossClusterTestCase {
+
+    @Before
+    public void checkSubqueryInFromCommandSupport() {
+        assumeTrue("Requires subquery in FROM command support", EsqlCapabilities.Cap.SUBQUERY_IN_FROM_COMMAND.isEnabled());
+    }
+
+    public void testSubquery() throws IOException {
+        setupClusters(3);
+
+        try (EsqlQueryResponse resp = runQuery("""
+            FROM (FROM logs-* metadata _index),(FROM *:logs-* metadata _index)
+            | SORT _index, id
+            """, randomBoolean())) {
+            var columns = resp.columns().stream().map(ColumnInfoImpl::name).toList();
+            assertThat(columns, hasItems("id", "tag", "v", "const", "_index"));
+            int constIndex = columns.indexOf("const");
+            int vIndex = columns.indexOf("v");
+            int indexIndex = columns.indexOf("_index");
+
+            List<List<Object>> values = getValuesList(resp);
+            assertThat(values, hasSize(30));
+            for (int i = 0; i < values.size(); i++) {
+                var row = values.get(i);
+                assertThat(row, hasSize(5));
+                assertNull(row.get(constIndex));
+                String indexName = (String) row.get(indexIndex);
+                if (i < 10) {
+                    assertEquals("cluster-a:logs-2", indexName);
```

Comment:
Maybe better to use the constants here? It's not entirely clear where these names come from, as they aren't part of this class. We could also make compound constants if necessary to make it more convenient.